joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -15,9 +15,9 @@ import java.nio.charset.StandardCharsets
 trait AstCreatorHelper(disableFileContent: Boolean)(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
 
   protected val globalNamespace: NewNamespaceBlock = globalNamespaceBlock()
-  private val tmpKeyPool                           = new IntervalKeyPool(first = 0, last = Long.MaxValue)
 
-  protected def getNewTmpName(prefix: String = "tmp"): String = s"$prefix${tmpKeyPool.next.toString}"
+  protected val tmpGen: FreshNameGenerator[String]   = FreshNameGenerator[String](i => s"tmp-$i")
+  protected val classGen: FreshNameGenerator[String] = FreshNameGenerator[String](i => s"anon-class-$i")
```

Comment:
Both of these naming schemas do not work. We currently have problems in go2cpg which uses the exact same naming schema.
Reason: The naming changes if surrounding unrelated code changes. So lets say you have an anon class at the end of a file nested in a function. You then add a new anon class somewhere at the start of that file and the name of the anon class at the end of the file would change.
Why is that a problem? Because it makes it harder to correlate unchanged code parts between different scans which we need in order to correlated findings between different scans.

So please use a naming schema which only uses an incremental counter as a last resort and use structural information (mostly nesting comes to mind) in order to compose a unique name.